subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ProgressEvents.groovy

```diff
@@ -62,7 +62,7 @@ class ProgressEvents implements ProgressListener {
                     // Display name should be mostly unique
                     // Ignore this check for TestOperationDescriptors as they are currently not unique when coming from different test tasks
                     if (!skipValidation && !(descriptor instanceof TestOperationDescriptor)) {
-                        if (descriptor.displayName == 'Configure settings' || descriptor.displayName == 'Configure build' || descriptor.displayName == 'Calculate task graph' || descriptor.displayName == 'Run tasks') {
+                        if (['Configure settings', 'Configure build', 'Calculate task graph', 'Run tasks', 'Apply plugin'].find { descriptor.displayName.startsWith(it) }) {
```

Comment:
we should make sure the operation has a unique display name (not name), which might be something like "applying plugin 'a' to project 'b'"